tools/codegen/model.py

```diff
@@ -197,8 +204,59 @@ def __post_init__(self) -> None:
                 "otherwise you will tickle a Python argument binding bug " \
                 "(which usually manifests itself as the result variable being undefined.)"
 
+        if self.structured:
+            assert self.func.kind() == SchemaKind.out, "Put structured field on the out= " \
+                "variant of a function"
+
 SchemaKind = Enum('SchemaKind', ('functional', 'inplace', 'out'))
 
+@dataclass(frozen=True)
+class NativeFunctionGroup:
```

Comment:
Thanks for creating this!

Reply:
Yeah, should backport this one to master ASAP